### PR TITLE
[BUGFIX] Fix Pauli.to_mat() for empty PauliSentence and add case handling of empty wires

### PR DESCRIPTION
**Pull request by Qottmann**
Fix https://github.com/PennyLaneAI/pennylane/issues/5187 and handle empty wire Identities since that is now allowed

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -473,6 +473,9 @@
   operators have a valid `pauli_rep` property.
   [(#5177)](https://github.com/PennyLaneAI/pennylane/pull/5177)
 
+* The matrix of an empty `PauliSentence` instance is now correct (all-zeros).
+  Further, matrices of empty `PauliWord` and `PauliSentence` instances can be turned to matrices now.
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:477]: This entry seems incorrect. We're not returning all zeros, just a 1x1 matrix.

**Review by mudit2812 — COMMENTED** [review]
Looks good to me! I'm totally on board with this decision. However, this is a change that I didn't really anticipate and I think a lot of other people might be initially surprised as well. I would definitely suggest adding documentation about this to `.to_mat()`.

File: doc/releases/changelog-dev.md (hunk 1 above)
Comment (Qottmann) [reply doc/releases/changelog-dev.md:477]:
```
>>> PauliSentence({}).to_mat(wire_order=[0, 1])
array([[0., 0., 0., 0.],
       [0., 0., 0., 0.],
       [0., 0., 0., 0.],
       [0., 0., 0., 0.]])
```

Comment (Qottmann) [reply doc/releases/changelog-dev.md:477]:
```
>>> PauliSentence({}).to_mat()
array([[0.]])
>>> PauliSentence({}).to_mat().shape
(1, 1)
```
If not all-zero matrix, how else would you call it?

Comment (mudit2812) [reply doc/releases/changelog-dev.md:477]: Ah yes I forgot about the case with wire order provided. Thanks!

File: pennylane/pauli/pauli_arithmetic.py [hunk 2]
```diff
@@ -539,7 +538,9 @@ class PauliSentence(dict):
     .. note::
 
         An empty :class:`~.PauliSentence` will be treated as the additive
-        identity (i.e ``0 * Identity()``).
+        identity (i.e ``0 * Identity()``). Its matrix is the all-zero matrix
+        (trivially the :math:`1\times 1` zero matrix when no `wire_order` is passed to
+        `PauliSentence({}).to_mat()`).
```
Comment (mudit2812) [inline pennylane/pauli/pauli_arithmetic.py:543]:
```suggestion
        (trivially the :math:`1\times 1` zero matrix when no ``wire_order`` is passed to
        ``PauliSentence({}).to_mat()``).
```

Comment (mudit2812) [reply pennylane/pauli/pauli_arithmetic.py:543]: I think a similar note for `PauliWord` would also be good.

**Comment by astralcai** [comment]
Hi, can I ask why the matrix of an empty `PauliWord` is identity but of an empty `PauliSentence` is a zero-matrix?

**Comment by Qottmann** [comment]
> Hi, can I ask why the matrix of an empty PauliWord is identity but of an empty PauliSentence is a zero-matrix?

Yes, that is a good question! An empty PauliWord constitutes the identity with respect to multiplication, whereas and empty PauliSentence consistutes the identity with respect to addition. I added some comments in the `note` of the PauliWord and PauliSentence class to make that more clear.

Does that help?

**Comment by astralcai** [comment]
> > Hi, can I ask why the matrix of an empty PauliWord is identity but of an empty PauliSentence is a zero-matrix?
> 
> Yes, that is a good question! An empty PauliWord constitutes the identity with respect to multiplication, whereas and empty PauliSentence consistutes the identity with respect to addition. I added some comments in the `note` of the PauliWord and PauliSentence class to make that more clear.
> 
> Does that help?

Thank you. That makes sense!

**Review by astralcai — APPROVED** [review]
Looks Good!

File: tests/pauli/test_pauli_arithmetic.py [hunk 3]
```diff
@@ -864,15 +854,21 @@ def test_iadd_ps_pw(self, ps, pw, res):
...
-    def test_to_mat_error_empty(self, ps, match):
-        """Test that an appropriate error is raised when an empty
-        PauliSentence or PauliWord is cast to matrix."""
-        with pytest.raises(ValueError, match=match):
-            ps.to_mat(wire_order=[])
+    PS_EMPTY_CASES = (
+        (PauliSentence({}), np.zeros((1, 1))),
+        (PauliSentence({PauliWord({}): 1.0}), np.ones((1, 1))),
+        (PauliSentence({PauliWord({}): 2.5}), 2.5 * np.ones((1, 1))),
+    )
```
Comment (astralcai) [inline tests/pauli/test_pauli_arithmetic.py:861]: Is there a test case for a `PauliSentence` that contains an empty `PauliWord` along with another non-empty `PauliWord` to make sure that it still works properly?

Comment (Qottmann) [reply tests/pauli/test_pauli_arithmetic.py:861]: Good call, will add! :)

Comment (Qottmann) [reply tests/pauli/test_pauli_arithmetic.py]: added ✅

**Review by mudit2812 — APPROVED** [review]

**Pull request merged**